src/backend/cod2.c

```diff
@@ -3996,89 +3995,86 @@ code *cdstreq(elem *e,regm_t *pretregs)
         {   need_DS = TRUE;
             srcregs |= mCX;
         }
-        c1a = codelem(e2,&srcregs,FALSE);
+        cdb.append(codelem(e2,&srcregs,FALSE));
     }
-    c1 = cat(c1,c1a);
 
-  /* now get pointer to lvalue (destination) in ES:DI                   */
-  dstregs = (config.exe & EX_flat) ? mDI : mES|mDI;
-  if (e1->Eoper == OPind)               /* if (*p = ..)                 */
-  {
+    // now get pointer to lvalue (destination) in ES:DI
+    regm_t dstregs = (config.exe & EX_flat) ? mDI : mES|mDI;
+    if (e1->Eoper == OPind)               // if (*p = ..)
+    {
         if (tyreg(e1->E1->Ety))
             dstregs = mDI;
-        c2 = cod2_setES(e1->E1->Ety);
-        c2 = cat(c2,scodelem(e1->E1,&dstregs,srcregs,FALSE));
-  }
-  else
-        c2 = cdrelconst(e1,&dstregs);
-  freenode(e1);
-
-  c3 = getregs((srcregs | dstregs) & (mLSW | mDI));
-  if (need_DS)
-  {     assert(!(config.exe & EX_flat));
-        c3 = gen1(c3,0x1E);                     /* PUSH DS              */
-        gen2(c3,0x8E,modregrm(3,SEG_DS,CX));    /* MOV DS,CX            */
-  }
-  if (numbytes <= REGSIZE * (6 + (REGSIZE == 4)))
-  {     while (numbytes >= REGSIZE)
+        cdb.append(cod2_setES(e1->E1->Ety));
+        cdb.append(scodelem(e1->E1,&dstregs,srcregs,FALSE));
+    }
+    else
+        cdb.append(cdrelconst(e1,&dstregs));
+    freenode(e1);
+
+    cdb.append(getregs((srcregs | dstregs) & (mLSW | mDI)));
+    if (need_DS)
+    {     assert(!(config.exe & EX_flat));
+        cdb.gen1(0x1E);                     // PUSH DS
+        cdb.gen2(0x8E,modregrm(3,SEG_DS,CX));    // MOV DS,CX
+    }
+    if (numbytes <= REGSIZE * (6 + (REGSIZE == 4)))
+    {
+        while (numbytes >= REGSIZE)
         {
-            c3 = gen1(c3,0xA5);         /* MOVSW                        */
-            code_orrex(c3, rex);
+            cdb.gen1(0xA5);         // MOVSW
+            code_orrex(cdb.last(), rex);
             numbytes -= REGSIZE;
         }
         //if (numbytes)
         //    printf("cdstreq numbytes %d\n",numbytes);
         while (numbytes--)
-            c3 = gen1(c3,0xA4);         /* MOVSB                        */
-  }
+            cdb.gen1(0xA4);         // MOVSB
+    }
     else
     {
 #if 1
-        unsigned remainder;
-
-        remainder = numbytes & (REGSIZE - 1);
+        unsigned remainder = numbytes & (REGSIZE - 1);
         numbytes /= REGSIZE;            // number of words
-        c3 = cat(c3,getregs_imm(mCX));
-        c3 = movregconst(c3,CX,numbytes,0);     // # of bytes/words
-        gen1(c3,0xF3);                          // REP
+        cdb.append(getregs_imm(mCX));
+        cdb.append(movregconst(CNIL,CX,numbytes,0));   // # of bytes/words
+        cdb.gen1(0xF3);                 // REP
```

Comment:
This acutally fixed a codegen bug, nice :).
[Issue 17220 – invalid code with -m32 -inline and struct that's size_t.sizeof x the size of an assigned enum value](https://issues.dlang.org/show_bug.cgi?id=17220)